In [1]:
#Lertas Giorgos 
#import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')
%matplotlib inline

### 1.Protein-protein interaction network

In [2]:
ppi=pd.read_csv('data Leskovec/bio-decagon-ppi.csv')
ppi

Gene 1  Gene 2
0       114787  375519
1       114787  285613
2       114787    7448
3       114787    4914
4       114787   51343
...        ...     ...
715607    5634    5636
715608    5635    5636
715609    5635    5631
715610    5636    5631
715611    8480    8487

[715612 rows x 2 columns]

In [3]:
ppi['Gene 1'].value_counts()

351       1986
1994      1493
6233      1228
2885      1106
4609      1049
          ... 
137133       1
138241       1
85349        1
56663        1
6345         1
Name: Gene 1, Length: 17544, dtype: int64

In [4]:
ppi['Gene 2'].value_counts()

4914      1825
653269    1431
728378    1392
653781    1266
7311      1132
          ... 
341116       1
153745       1
54979        1
259239       1
340571       1
Name: Gene 2, Length: 17472, dtype: int64

The network has 17544 different proteins in the first column as Gene 1 and 17472 in the second as Gene 2. Also, it has 715612  physical interactions or observations.

### 2.Drug-target protein associations

In [5]:
targets=pd.read_csv('data Leskovec/bio-decagon-targets.csv')
targets

STITCH   Gene
0      CID000003488   1559
1      CID000003488   8647
2      CID000077992   3351
3      CID000077992   3350
4      CID000077992   3352
...             ...    ...
18685  CID000005152   8484
18686  CID000005152  81491
18687  CID000005152  83551
18688  CID000005152    680
18689  CID000005152  11255

[18690 rows x 2 columns]

In [6]:
targets.STITCH.value_counts()

CID000000271    2840
CID005329102     253
CID000002801     206
CID000002818     206
CID000004543     199
                ... 
CID000003325       1
CID000004046       1
CID000004499       1
CID000003878       1
CID000004723       1
Name: STITCH, Length: 284, dtype: int64

In [7]:
targets.Gene.value_counts()

3356      94
3357      89
151       87
3274      86
150       86
          ..
794        1
2811       1
10975      1
2779       1
133121     1
Name: Gene, Length: 3648, dtype: int64

This association has 284 diffent drugs as STITCH (Search Tool for InTeractions of CHemicals) and 3648 different proteins. The matrix has 18690 instances.

### 3.Drug-target protein associations culled from several curated databases

In [8]:
targets_all=pd.read_csv('data Leskovec/bio-decagon-targets-all.csv')
targets_all

STITCH   Gene
0       CID000060752   3757
1       CID006918155   2908
2       CID103052762   3359
3       CID023668479   1230
4       CID000028864   1269
...              ...    ...
131029  CID000092721   3426
131030  CID000092721   8858
131031  CID000092721  10942
131032  CID100115355   3242
131033  CID100115355  84842

[131034 rows x 2 columns]

In [9]:
targets_all.STITCH.value_counts()

CID000000753    3812
CID100000753    3812
CID100000312    2921
CID000023994    2867
CID100023994    2866
                ... 
CID000077993       1
CID100005718       1
CID100077993       1
CID100004493       1
CID116129617       1
Name: STITCH, Length: 1774, dtype: int64

In [10]:
targets_all.Gene.value_counts()

3356     438
3358     422
3357     420
3350     416
150      412
        ... 
5866       1
55619      1
2309       1
324        1
3592       1
Name: Gene, Length: 7795, dtype: int64

This association has more drugs than before, 1774 drugs as STITCH (Search Tool for InTeractions of CHemicals) and 7795 different proteins. Also, the dataframe has 131034 instances.

### 4.Polypharmacy side effects in the form of (drug A, side effect type, drug B) triples.

In [11]:
combo=pd.read_csv('data Leskovec/bio-decagon-combo.csv')
combo

STITCH 1      STITCH 2 Polypharmacy Side Effect  \
0        CID000002173  CID000003345                 C0151714   
1        CID000002173  CID000003345                 C0035344   
2        CID000002173  CID000003345                 C0004144   
3        CID000002173  CID000003345                 C0002063   
4        CID000002173  CID000003345                 C0004604   
...               ...           ...                      ...   
4649436  CID000003461  CID000003954                 C0149871   
4649437  CID000003461  CID000003954                 C0035410   
4649438  CID000003461  CID000003954                 C0043096   
4649439  CID000003461  CID000003954                 C0003962   
4649440  CID000003461  CID000003954                 C0038999   

                   Side Effect Name  
0                   hypermagnesemia  
1        retinopathy of prematurity  
2                       atelectasis  
3                         alkalosis  
4                         Back Ache  
...                             ...  
4649436        deep vein thromboses  
4649437              rhabdomyolysis  
4649438              loss of weight  
4649439                     ascites  
4649440                     bulging  

[4649441 rows x 4 columns]

In [12]:
combo['STITCH 1'].value_counts()

CID000000853    68416
CID000002083    65709
CID000001983    62959
CID000002771    55784
CID000002662    54811
                ...  
CID000002673       12
CID000150610       10
CID005282044       10
CID000002551        7
CID000051634        6
Name: STITCH 1, Length: 616, dtype: int64

In [13]:
combo['STITCH 2'].value_counts()

CID000005732    56959
CID000005090    51484
CID000062924    49715
CID000004900    49484
CID000054454    45713
                ...  
CID000000564       14
CID000002800       13
CID000039860       12
CID000002551        9
CID000005482        8
Name: STITCH 2, Length: 634, dtype: int64

In [14]:
combo['Polypharmacy Side Effect'].value_counts()

C0020649    28568
C0002871    27006
C0013404    26037
C0027497    25190
C0032285    24430
            ...  
C0013336        2
C0162678        2
C0011593        2
C0014935        2
C0034494        1
Name: Polypharmacy Side Effect, Length: 1317, dtype: int64

In [15]:
combo['Side Effect Name'].value_counts()

arterial pressure NOS decreased    28568
anaemia                            27006
Difficulty breathing               26037
nausea                             25190
neumonia                           24430
                                   ...  
neurofibromatosis                      2
skin abrasion                          2
hypogonadotropic hypogonadism          2
dwarfism                               2
rabies                                 1
Name: Side Effect Name, Length: 1317, dtype: int64

In this dataframe, we can see 616 diffent drugs as STITCH 1 and 634 diffent drugs as STITCH 2. Also, Polypharmacy Side Effect and Side Effect Name are 1317 and the observations are 4649441

### 5.Side effects of individual drugs in the form of (drug A, side effect type) tuples.

In [16]:
mono=pd.read_csv('data Leskovec/bio-decagon-mono.csv')
mono

STITCH Individual Side Effect  \
0       CID003062316               C1096328   
1       CID003062316               C0162830   
2       CID003062316               C1611725   
3       CID003062316               C0541767   
4       CID003062316               C0242973   
...              ...                    ...   
174972  CID005282226               C1096278   
174973  CID005282226               C0272412   
174974  CID005282226               C0860013   
174975  CID005282226               C0853225   
174976  CID005282226               C1112371   

                                Side Effect Name  
0                    central nervous system mass  
1                      Photosensitivity reaction  
2                   leukaemic infiltration brain  
3                 platelet adhesiveness abnormal  
4                        Ventricular dysfunction  
...                                          ...  
174972             anterior chamber inflammation  
174973                           splenic abscess  
174974             hepatitis b antibody positive  
174975  international normalised ratio increased  
174976             Anterior chamber pigmentation  

[174977 rows x 3 columns]

In [17]:
mono.STITCH.value_counts()

CID000068740    1550
CID000004674    1515
CID000005538    1229
CID000000444    1186
CID000005064    1045
                ... 
CID000072938      29
CID000004913      25
CID000005482      19
CID000002551      10
CID000003278       5
Name: STITCH, Length: 639, dtype: int64

In [18]:
mono['Individual Side Effect'].value_counts()

C0877211    301
C1167981    279
C0856054    278
C0040440    276
C0700361    275
           ... 
C1535944      1
C2363894      1
C0195488      1
C0861100      1
C0577598      1
Name: Individual Side Effect, Length: 10184, dtype: int64

In [19]:
mono['Side Effect Name'].value_counts()

general physical health deterioration    301
hypoaesthesia                            279
mental status changes                    278
tooth extraction                         276
emotional distress                       275
                                        ... 
hypertrophic osteoarthropathy              1
cryptococcus antigen positive              1
Eczema eyelids                             1
Protein total decreased                    1
immunisation                               1
Name: Side Effect Name, Length: 9702, dtype: int64

This table has 174977 rows and 639 drugs as STITCH. Also, the columns Individual Side Effect and Side Effect Name have 
10184 and 9702 observations respectively.

### 6.Side effect categories

In [20]:
effectcategories=pd.read_csv('data Leskovec/bio-decagon-effectcategories.csv')
effectcategories

Side Effect          Side Effect Name  \
0      C0017152      gastric inflammation   
1      C0027858                   neuroma   
2      C0041466                   Typhoid   
3      C0032807  Post thrombotic syndrome   
4      C0033860                 psoriasis   
..          ...                       ...   
556    C0001816                   agnosia   
557    C0302592   carcinoma of the cervix   
558    C0270629          epidural abscess   
559    C0038220        status epilepticus   
560    C0042996                  vulvitis   

                               Disease Class  
0            gastrointestinal system disease  
1                            benign neoplasm  
2               bacterial infectious disease  
3              cardiovascular system disease  
4               integumentary system disease  
..                                       ...  
556  developmental disorder of mental health  
557                                   cancer  
558                   nervous system disease  
559                   nervous system disease  
560              reproductive system disease  

[561 rows x 3 columns]

In [21]:
effectcategories['Side Effect'].value_counts()

C0024110    1
C0033845    1
C0035304    1
C0007177    1
C0006826    1
           ..
C0005695    1
C0162423    1
C0033578    1
C0024198    1
C0013537    1
Name: Side Effect, Length: 561, dtype: int64

In [22]:
effectcategories['Side Effect Name'].value_counts()

amyloidosis          1
histoplasmosis       1
acute bronchitis     1
bone sarcoma         1
thrombocytopenia     1
                    ..
chronic hepatitis    1
mucositis oral       1
anogenital warts     1
Collagen disease     1
Rubella              1
Name: Side Effect Name, Length: 561, dtype: int64

In [23]:
effectcategories['Disease Class'].value_counts().count()

37

In [24]:
effectcategories['Disease Class'].value_counts().count()

37

This dataframe has 561 observations, which they are uniquely 'Side Effect' ids and 'Side Effect Name' as column names, and it has 37 different names in 'Disease Class'.